sdk/samples/azidentity/example_SDKV1_test.go

```diff
@@ -0,0 +1,143 @@
+// with azidentity package from the Azure SDK for Go V2.
+// The adapter in the azidext package provides a simple way to integrate azidentity credentials
+// as authorizers for the V1 code base.
+
+const (
+	groupName = "samplegroup"
+)
+
+// Environment variables required for EnvironmentCredential to work and/or DefaultAzureCredential
+var (
+	clientID       = os.Getenv("AZURE_CLIENT_ID")
+	clientSecret   = os.Getenv("AZURE_CLIENT_SECRET")
+	subscriptionID = os.Getenv("AZURE_SUBSCRIPTION_ID")
+	tenantID       = os.Getenv("AZURE_TENANT_ID")
+)
+
+var (
+	location  = os.Getenv("AZURE_LOCATION")
+	userAgent = "azidentitysample"
+)
+
+// ExampleGroupsClientWithDefaultAzureCredential for using the DefaultAzureCredential through the NewDefaultAzureCredentialAdapter and assigning the credential to the
+// SDK V1 authorizer.
+// NewDefaultAzureCredentialAdapter should be used to replace auth.NewAuthorizerFromEnvironment(). DefaultAzureCredential, similarly to NewAuthorizerFromEnvironment, checks for
+// environment variables that can construct ClientSecretCredentials, ClientCertificateCredentials, UsernamePasswordCredentials, ManagedIdentityCredentials and AzureCLICredentials.
+func ExampleGroupsClientWithDefaultAzureCredential() {
+	groupsClient := resources.NewGroupsClient(subscriptionID)
+	// call azidext.NewDefaultAzureCredentialAdapter in order to get an authorizer with a DefaultAzureCredential
+	// leave azidext.DefaultAzureCredentialOptions as nil to get the default scope for management APIs.
+	// The default scope is: https://management.azure.com//.default.
+	// NOTE: Scopes define the set of resources and permissions that the credential will have assigned to it.
+	// 		 To read more about scopes, see: https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-permissions-and-consent
+	a, err := azidext.NewDefaultAzureCredentialAdapter(nil)
+	if err != nil {
+		panic("failed to get credential")
+	}
+	groupsClient.Authorizer = a
+	// use the groups client with the azidentity credential in the authorizer
+	group, err := groupsClient.CreateOrUpdate(
+		context.Background(),
+		groupName,
+		resources.Group{
+			Location: to.StringPtr(location),
+		})
+	if err != nil {
+		panic(err)
+	}
+	fmt.Println(*group.Name)
+	// Output:
+	// samplegroup
+}
+
+// ExampleGroupsClientWithClientSecretCredential for using the ClientSecretCredential with the NewAzureIdentityCredentialAdapter and assigning the credential to the
+// SDK V1 authorizer.
+// NewAzureIdentityCredentialAdapter can take any credential type defined in azidentity and convert it to an authorizer that is compatible with the Azure SDK for Go
+// V1 implementation. For a list of the credentials that azidentity includes, please see: https://github.com/Azure/azure-sdk-for-go/tree/master/sdk/azidentity.
+// NewClientSecretCredential can be used in place of any authorizer that supports client credentials.
```

Comment:
The closest replacement for v1 is auth.NewClientCredentialsConfig().  I think this helps highlight the improvements in azidentity.

Reply:
Done